librz/bin/format/mach0/mach0_common.c

```diff
@@ -0,0 +1,250 @@
+// SPDX-FileCopyrightText: 2023 Florian Märkl <info@florianmaerkl.de>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_types.h>
+#include "mach0_defines.h"
+
+/**
+ * String representation of MACH0_PLATFORM_* values, e.g. from LC_BUILD_VERSION load command
+ */
+RZ_API RZ_NONNULL const char *rz_mach0_platform_to_string(ut32 platform) {
+	switch (platform) {
+	case MACH0_PLATFORM_MACOS:
+		return "macOS";
+	case MACH0_PLATFORM_IOS:
+		return "iOS";
+	case MACH0_PLATFORM_TVOS:
+		return "tvOS";
+	case MACH0_PLATFORM_WATCHOS:
+		return "watchOS";
+	case MACH0_PLATFORM_BRIDGEOS:
+		return "bridgeOS";
+	case MACH0_PLATFORM_IOS_MAC:
+		return "Mac Catalyst";
+	case MACH0_PLATFORM_IOS_SIMULATOR:
+		return "iOS Simulator";
+	case MACH0_PLATFORM_TVOS_SIMULATOR:
+		return "tvOS Simulator";
+	case MACH0_PLATFORM_WATCHOS_SIMULATOR:
+		return "watchOS Simulator";
+	case MACH0_PLATFORM_DRIVERKIT:
+		return "driverKit";
+	case MACH0_PLATFORM_VISIONOS:
+		return "visionOS";
+	case MACH0_PLATFORM_VISIONOS_SIMULATOR:
+		return "visionOS Simulator";
+	default:
+		return "unknown";
+	}
+}
+
+/**
+ * String representation of CPU_TYPE_* values
+ */
+RZ_API RZ_NONNULL const char *rz_mach0_cputype_to_string(ut32 cputype) {
+	switch (cputype) {
+	case CPU_TYPE_VAX:
+		return "vax";
+	case CPU_TYPE_MC680x0:
+		return "mc680x0";
+	case CPU_TYPE_I386:
+	case CPU_TYPE_X86_64:
+		return "x86";
+	case CPU_TYPE_MC88000:
+		return "mc88000";
+	case CPU_TYPE_MC98000:
+		return "mc98000";
+	case CPU_TYPE_HPPA:
```

Comment:
I didn't knew it supported HPPA and SPARC

Reply:
I think these still come from CMU Mach